pkg/operator/baremetal_test.go

```diff
@@ -71,3 +95,31 @@ func TestCreateMariadbPasswordSecret(t *testing.T) {
 		t.Logf("First Mariadb password is being preserved over re-creation as expected.")
 	}
 }
+
+func TestGetBaremetalProvisioningConfig(t *testing.T) {
+	u := &unstructured.Unstructured{Object: map[string]interface{}{}}
+	if err := yaml.Unmarshal([]byte(yamlContent), &u); err != nil {
+		t.Errorf("failed to unmarshall input yaml content:%v", err)
+	}
+	dynamicClient := fakedynamic.NewSimpleDynamicClient(runtime.NewScheme(), u)
+	baremetalConfig := getBaremetalProvisioningConfig(dynamicClient, "test")
+	if baremetalConfig.ProvisioningInterface != expectedProvisioningInterface ||
+		baremetalConfig.ProvisioningIp != expectedProvisioningIP ||
+		baremetalConfig.ProvisioningNetworkCIDR != expectedProvisioningNetworkCIDR ||
+		baremetalConfig.ProvisioningDHCPExternal != expectedProvisioningDHCPExternal ||
+		baremetalConfig.ProvisioningDHCPRange != expectedProvisioningDHCPRange {
+		t.Errorf("failed getBaremetalProvisioningConfig. One or more BaremetalConfig items do not match the expected config.")
```

Comment:
can you render here the expected vs provisioned?

Reply:
Done.